[WFLY-17070][WFLY-17053][WFLY-16867] Drop source transformation for the Agroal subsystem

yersan opened the pull request:
Jira issue: https://issues.redhat.com/browse/WFLY-17070


In addition, amends minor fixes for:
https://issues.redhat.com/browse/WFLY-17053
https://issues.redhat.com/browse/WFLY-16867

yersan committed:
[WFLY-17070] Drop source transformation for the Agroal subsystem

Jira issue: https://issues.redhat.com/browse/WFLY-17070

bstansberry reviewed (changes requested):
Thanks @yersan 

Please handle this TODO:

https://github.com/wildfly/wildfly/blob/main/pom.xml#L1988-L1989

Bonus points for fixing mail and jaxrs that have already been migrated where I forgot to ask for this in code review.

yersan committed:
[WFLY-17053] Remove wildfly-mail from ee8.build.profile profile

- This module is purely based on jakarta.* package names, so it no longer needs to be skipped when we are only interested in WildFly Preview builds.

yersan committed:
[WFLY-16867] Remove wildfly-jaxrs from ee8.build.profile profile

- This module is purely based on jakarta.* package names, so it no longer needs to be skipped when we are only interested in WildFly Preview builds.

yersan commented:
Good catch @bstansberry , thanks for reviewing it

bstansberry commented:
Thanks, @yersan
